src/lava/magma/runtime/message_infrastructure/CMakeLists.txt

```diff
@@ -69,13 +68,48 @@ if(GRPC_CHANNEL)
     "message_infrastructure/csrc/channel/grpc/grpc_port.cc"
     "message_infrastructure/csrc/channel/grpc/grpc_channel.cc"
     ${grpc_proto_srcs}
-    ${grpc_srcs}
-  )
-  add_library(message_infrastructure SHARED ${MESSAGE_INFRASTRUCTURE_SRCS} ${GRPC_CHANNEL_SRCS})
+    ${grpc_srcs})
+endif()
+
+option(DDS_CHANNEL "Message library supports DDS Channel" OFF)
+option(FASTDDS_ENABLE "enable FastDDS" OFF)
+option(CycloneDDS_ENABLE "enable CycloneDDS" OFF)
+
+if(DDS_CHANNEL)
+  set(COMMON_DDS_DESTINATION "${CMAKE_BINARY_DIR}/install")
+  set(COMMON_DDS_INC "${COMMON_DDS_DESTINATION}/include")
+  set(DDS_CHANNEL_SRCS
+    "message_infrastructure/csrc/channel/dds/dds.cc"
+    "message_infrastructure/csrc/channel/dds/dds_channel.cc")
+
+  if(CycloneDDS_ENABLE)
+    set(CycloneDDS_SRC "")
+  endif()
+
+  if(FASTDDS_ENABLE)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/fastdds.cmake)
+    add_definitions(-DFASTDDS)
+    set(FASTDDS_SRC
+      "message_infrastructure/csrc/channel/dds/fast_dds.cc"
+      "message_infrastructure/csrc/channel/dds/protos/fast_dds/metadata.cc"
+      "message_infrastructure/csrc/channel/dds/protos/fast_dds/metadataPubSubTypes.cc"
+      ${DDS_CHANNEL_SRCS})
+  endif()
+
+  if(FASTDDS_ENABLE OR CycloneDDS_ENABLE)
+    add_definitions(-DDDS_CHANNEL)
```

Comment:
Seems there is no need to judge FASTDDS_ENABLE or CycloneDDS_ENABLE to add DDS_CHANNEL definition.

Reply:
I'll add an exception for that in cmake.